dlc-manager/src/manager.rs

```diff
@@ -976,16 +1017,25 @@ where
                 // not much to be done apart from possibly extracting a fraud
                 // proof but ideally it should be handled.
                 self.blockchain.send_transaction(&cet)?;
-            }
 
-            let closed_contract = ClosedContract {
-                signed_contract: contract.clone(),
-                attestations: attestations.iter().map(|x| x.1.clone()).collect(),
-                cet_index: range_info.cet_index,
-            };
+                let preclosed_contract = PreClosedContract {
+                    signed_contract: contract.clone(),
+                    attestations: attestations.iter().map(|x| x.1.clone()).collect(),
+                    cet_index: range_info.cet_index,
+                };
 
-            self.store
-                .update_contract(&Contract::Closed(closed_contract))?;
+                self.store
+                    .update_contract(&Contract::PreClosed(preclosed_contract))?;
+            } else {
```

Comment:
I think the ideal would be to have a case for `1 <= confirmations < 6` in which it's pre-closed, and then this else.